modules/apps/29-fee/keeper/grpc_query.go

```diff
@@ -50,13 +50,11 @@ func (k Keeper) IncentivizedPackets(ctx context.Context, req *types.QueryIncenti
 }
 
 // IncentivizedPacket implements the Query/IncentivizedPacket gRPC method
-func (k Keeper) IncentivizedPacket(goCtx context.Context, req *types.QueryIncentivizedPacketRequest) (*types.QueryIncentivizedPacketResponse, error) {
+func (k Keeper) IncentivizedPacket(ctx context.Context, req *types.QueryIncentivizedPacketRequest) (*types.QueryIncentivizedPacketResponse, error) {
 	if req == nil {
 		return nil, status.Error(codes.InvalidArgument, "empty request")
 	}
 
-	ctx := sdk.UnwrapSDKContext(goCtx).WithBlockHeight(int64(req.QueryHeight))
```

Comment:
Note that the understanding of `WithBlockHeight` when this was done by previous implementers was incorrect. This does not load the ibc store at the requested height only overwrite the context field at the given moment. 

At this point the multistore instance available to the request context has already been loaded by baseapp. The intended functionality for cosmos-sdk can be achieved by passing the `x-cosmos-block-height` grpc header via metadata